src/waterlinked_dvl/waterlinked_dvl_tcp.cpp

```diff
@@ -190,6 +192,18 @@ void WaterlinkedDvlTcp::f_parse_json_v2(Json::Value root) {
                 }
                 m_twist_publisher.publish(twist_msg);
             }
+            //Raw Twist
+            geometry_msgs::TwistWithCovarianceStamped twist_msg;
+            twist_msg.header = msg.header;
+            twist_msg.twist.twist.linear.x = msg.report.vx;
+            twist_msg.twist.twist.linear.y = msg.report.vy;
+            twist_msg.twist.twist.linear.z = msg.report.vz;
+
+            if (m_velocity_covariance.size() == twist_msg.twist.covariance.size()) {
```

Comment:
covariance should from the transducer report no a userdefined value. the mapping of the values are wrong.

Reply:
json_v2 format do not have covariance matrix.
dvl/twist_raw can only then be derived from json_v3 format.